demo/c/pv_speaker_demo.c

```diff
@@ -0,0 +1,227 @@
+        return NULL;
+    }
+
+    if (header.num_channels != 1) {
+        fclose(file);
+        fprintf(stderr, "WAV file must have a single channel (MONO)\n");
+        return NULL;
+    }
+
+    *sample_rate = header.sample_rate;
+
+    *bits_per_sample = header.bits_per_sample;
+
+    uint32_t bytes_per_sample = header.bits_per_sample / 8;
+    *num_samples = header.subchunk2_size / bytes_per_sample;
+
+    void *pcm_data = malloc(header.subchunk2_size);
+    if (!pcm_data) {
+        perror("Memory allocation failed");
+        fclose(file);
+        return NULL;
+    }
+
+    fread(pcm_data, header.subchunk2_size, 1, file);
+
+    fclose(file);
+
+    return pcm_data;
+}
+
+int main(int argc, char *argv[]) {
+    const char *input_wav_path = NULL;
+    int32_t device_index = -1;
+
+    int c;
+    while ((c = getopt_long(argc, argv, "si:d:", long_options, NULL)) != -1) {
+        switch (c) {
+            case 's':
+                show_audio_devices();
+                return 0;
+            case 'i':
+                input_wav_path = optarg;
+                break;
+            case 'd':
+                device_index = (int32_t) strtol(optarg, NULL, 10);
+                break;
+            default:
+                exit(1);
+        }
+    }
+
+    if (!input_wav_path) {
+        print_usage(argv[0]);
+        exit(1);
+    }
+
+    signal(SIGINT, interrupt_handler);
+    fprintf(stdout, "pv_speaker version: %s\n", pv_speaker_version());
+
+    uint32_t num_samples, sample_rate;
+    uint16_t bits_per_sample;
+    void *pcm = read_wav_file(input_wav_path, &num_samples, &sample_rate, &bits_per_sample);
+
+    fprintf(stdout, "Initializing pv_speaker...\n");
+    const int32_t frame_length = 512;
+    pv_speaker_t *speaker = NULL;
+    pv_speaker_status_t status = pv_speaker_init(
+            sample_rate,
+            frame_length,
+            bits_per_sample,
+            device_index,
+            10,
+            &speaker);
+    if (status != PV_SPEAKER_STATUS_SUCCESS) {
+        fprintf(stderr, "Failed to initialize device with %s.\n", pv_speaker_status_to_string(status));
+        exit(1);
+    }
+
+    pv_speaker_set_debug_logging(speaker, true);
+
+    const char *selected_device = pv_speaker_get_selected_device(speaker);
+    fprintf(stdout, "Selected device: %s.\n", selected_device);
+
+    status = pv_speaker_start(speaker);
+    if (status != PV_SPEAKER_STATUS_SUCCESS) {
+        fprintf(stderr, "Failed to start device with %s.\n", pv_speaker_status_to_string(status));
+        exit(1);
+    }
+
+    fprintf(stdout, "Playing audio...\n");
+    if (pcm) {
+        char *pcmData = (char *) pcm;
+        for (int i = 0; i < num_samples; i += frame_length) {
+            bool is_last_frame = i + frame_length >= num_samples;
+            int32_t last_frame_length = num_samples - i;
+
+            status = pv_speaker_write(
+                    speaker,
+                    is_last_frame ? last_frame_length : frame_length,
```

Comment:
nipicking, but you can change it to 
```suggestion
                    is_last_frame ? num_samples - i : frame_length,
```
and avoid evaluating unnecessary value